test/python/test_sampler.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+from typing import Any
+
+import pytest
+from qiskit import QuantumCircuit
+from qiskit.circuit.library import RealAmplitudes
+from qiskit.primitives import SamplerResult
+
+from mqt.ddsim.sampler import Sampler
+
+
+@pytest.fixture()
+def sampler() -> Sampler:
+    """The sampler fixture for the tests in this file."""
+    return Sampler()
+
+
+@pytest.fixture()
+def circuits() -> list[QuantumCircuit]:
+    """The circuit lists fixture for the tests in this file."""
+
+    bell_1 = QuantumCircuit(2)
+    bell_1.h(0)
+    bell_1.cx(0, 1)
+    bell_1.measure_all()
+
+    bell_2 = QuantumCircuit(2)
+    bell_2.h(0)
+    bell_2.cx(0, 1)
+    bell_2.x(1)
+    bell_2.measure_all()
+
+    param_qc_1 = RealAmplitudes(num_qubits=2, reps=2)
+    param_qc_1.measure_all()
+
+    param_qc_2 = RealAmplitudes(num_qubits=2, reps=3)
+    param_qc_2.measure_all()
+
+    return [bell_1, bell_2, param_qc_1, param_qc_2]
+
+
+def compare_probs(
+    prob: list[dict[Any, float]] | dict[Any, float],
+    target: list[dict[Any, float]] | dict[Any, float],
+    tolerance: float = 0.045,
```

Comment:
Hm. 5Mil Shots is quite a lot. I am rather surprised that this does not yield the desired accuracy.
Do you see any way to adjust the tests themselves so that they are not as flaky but can use a reasonable tolerance?

Reply:
Found that 5 mil wasn't necessary, it was just that one of the more complex tests still had the shots number set to 1024.
I set shots = 32796 for all the tests and it is working well now